test/src/test/java/org/neo4j/ogm/persistence/types/ClassHierarchiesIntegrationTest.java

```diff
@@ -704,8 +704,10 @@ public void shouldReadHierarchy2() {
         Collection<Female> females = session.loadAll(Female.class);
         Collection<Bloke> blokes = session.loadAll(Bloke.class);
 
-        assertThat(entities).hasSize(3);
-        assertThat(entities.containsAll(Arrays.asList(daniela, michal, adam))).isTrue();
+        // to be consistent with the fact that non annotated abstract classes are not managed,
```

Comment:
Two things 

- Which change causes this change in behaviour?

- I am a bit worried about this _invisible_ change in behaviour, generally people shouldn't query by non-annotated abstract class, can we put at least a warning when they do? Or maybe an exception?

Reply:
This was to be consistent with other data access that ignore non annotated abstract classes (save, count, ...). 
I'd favor fail fast too. But not sure it is a good idea to introduce an exception when people use abstract classes now.

Reply:
Will add a warning in the logs